### PR TITLE
Fixes leg day

### LINKED ISSUE #32474: Roundstart freezing issue
At the start of a round on either bagil or sybil, most recently round 78433 on bagil, people are unable to move.

The effects are:
1. Most importantly, people who joined at roundstart are completely unable to move. Latejoiners are able to move normally.
2. Their mobs are not selectable in the mouse2 menu
3. Their name is null of null type (not a string containing null). Pastebin of VV available https://pastebin.com/YyscAxKS , though i've already tried varediting things like name and mouse opacity stuff

image

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #32474 

@MrStonedOne please

### CODE REVIEW

**Comment by MrStonedOne** [comment]
'cuseme bitch:

https://github.com/tgstation/tgstation/commit/ec8c42bbff4077bfe7d4df1ae0b0f54aae566f52

**Comment by Iamgoofball** [comment]
what is this wizardry and how does this fix skipping leg day?

**Comment by MrStonedOne** [comment]
@cyberboss defiled my pretty maiden and gave it a bug

**Comment by AnturK** [comment]
Oh sorry MSO, didn't see the blame. Oh boy and i even merged that, now i feel worse.

**Comment by kevinz000** [comment]
wtf

**Review by Cyberboss — DISMISSED** [review]

**Comment by RandomMarine** [comment]
Leg day happened with this testmerged on Sybil round 78724.

**Comment by AnturK** [comment]
Hmm

**Comment by psykzz** [comment]
Why can't this use the sorts already defined in https://github.com/tgstation/tgstation/blob/master/code/__HELPERS/sorts/__main.dm ?

**Comment by MrStonedOne** [comment]
They have overhead.

A sorted insert is a special kind of sort for inserting an item into an already sorted list.

Resorting the list every insert would be a massive waste of time and cpu

**Comment by psykzz** [comment]
I thought the timSort was meant to be optimised for a list that was partially sorted, although we should still add that func into the `__HELPERS/sorts` so it can be reused at some point.

**Comment by psykzz** [comment]
@AnturK instead of force pushing and rewriting history why not just squash the merge?

**Comment by AnturK** [comment]
Mostly because i didn't feel like guessing which commit is testmerged at the moment.

**Comment by dhostak** [comment]
Didn't work as of round ID 78757

**Review by Cyberboss — DISMISSED** [review]

**Comment by MrStonedOne** [comment]
I had some issues getting the test merge to update on sybil. i'll retry.

File: code/controllers/subsystem/timer.dm [hunk 1]
```diff
@@ -259,11 +259,11 @@ SUBSYSTEM_DEF(timer)
 			var/datum/timedevent/Last = ctts[cttl]
 			if(Last.timeToRun >= timeToRun)
 				ctts += src
-			else if(cttl > 1)
-				for(var/I in cttl to 1)
-					var/datum/timedevent/E = ctts[I]
+			else
+				for(var/i=cttl,i>=1,i--)
```
Comment (duncathan) [inline code/controllers/subsystem/timer.dm]: `for(var/i in cttl to 1 step -1)` will fix this in the same way while maintaining the performance benefit of using ranges

**Review by duncathan — DISMISSED** [review]

**Pull request merged**